src/core/tsi/ssl_transport_security.cc

```diff
@@ -2066,27 +2063,6 @@ tsi_result tsi_create_ssl_client_handshaker_factory_with_options(
   }
 
   do {
```

Comment:
This is irrelevant to this change, but in regards to the pattern the code is using: 
```
do {
  // ...
} while (false);
```
it seems the only reason for using this pattern is to save space for doing `tsi_ssl_handshaker_factory_unref` in multiple places. But I think it sort of affects the readability, so I am not sure what's the best way in normal C++ style guide to handle this case. 

Might need @markdroth 's opinion on this...